numerics/polynomial_evaluators_body.hpp

```diff
@@ -118,104 +120,150 @@ struct InternalEstrinEvaluator<Value, Argument, degree, low, 0> {
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 0>::Evaluate(
+template<typename Value, typename Argument, int degree, int low, bool fma>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, low, 0, fma>::Evaluate(
     Coefficients const& coefficients,
     Argument const& argument,
     ArgumentSquares const& argument_squares) {
   return std::get<low>(coefficients);
 }
 
-template<typename Value, typename Argument, int degree, int low>
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 1>::EvaluateDerivative(
-    Coefficients const& coefficients,
-    Argument const& argument,
-    ArgumentSquares const& argument_squares) {
-  return low * std::get<low>(coefficients) +
-         argument * (low + 1) * std::get<low + 1>(coefficients);
+template<typename Value, typename Argument, int degree, int low, bool fma>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, low, 1, fma>::
+    EvaluateDerivative(Coefficients const& coefficients,
+                       Argument const& argument,
+                       ArgumentSquares const& argument_squares) {
+  auto const& x = argument;
+  auto const& a = (low + 1) * std::get<low + 1>(coefficients);
+  auto const& b = low * std::get<low>(coefficients);
+  if constexpr (fma) {
+    return FusedMultiplyAdd(a, x, b);
+  } else {
+    return a * x + b;
+  }
 }
 
-template<typename Value, typename Argument, int degree, int low>
-Derivative<Value, Argument, low>
-InternalEstrinEvaluator<Value, Argument, degree, low, 0>::EvaluateDerivative(
-    Coefficients const& coefficients,
-    Argument const& argument,
-    ArgumentSquares const& argument_squares) {
+template<typename Value, typename Argument, int degree, int low, bool fma>
+FORCE_INLINE(inline) Derivative<Value, Argument, low>
+InternalEstrinEvaluator<Value, Argument, degree, low, 0, fma>::
+    EvaluateDerivative(Coefficients const& coefficients,
+                       Argument const& argument,
+                       ArgumentSquares const& argument_squares) {
   return low * std::get<low>(coefficients);
 }
 
 template<typename Value, typename Argument, int degree>
-Value EstrinEvaluator<Value, Argument, degree>::Evaluate(
+template<bool fma>
+FORCE_INLINE(inline) Value EstrinEvaluator<Value, Argument, degree>::Evaluate(
     Coefficients const& coefficients,
     Argument const& argument) {
   using InternalEvaluator = InternalEstrinEvaluator<Value,
                                                     Argument,
                                                     degree,
                                                     /*low=*/0,
-                                                    /*subdegree=*/degree>;
+                                                    /*subdegree=*/degree,
+                                                    fma>;
```

Comment:
There is a case to be made for putting `fma` immediately after `degree` since it's an external parameters, and `low` and `subdegree` are just recursion mess.